sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3979,23 +3982,41 @@ pi_result piEnqueueMemUnmap(pi_queue Queue, pi_mem MemObj, void *MappedPtr,
 
   // TODO: Level Zero is missing the memory "mapping" capabilities, so we are
   // left to doing copy (write back to the device).
+  // See https://gitlab.devtools.intel.com/one-api/level_zero/issues/293. //
+  // INTEL
   //
   // NOTE: Keep this in sync with the implementation of
   // piEnqueueMemBufferMap/piEnqueueMemImageMap.
   _pi_mem::Mapping MapInfo = {};
   if (pi_result Res = MemObj->removeMapping(MappedPtr, MapInfo))
     return Res;
 
-  ZE_CALL(zeCommandListAppendMemoryCopy(
-      ZeCommandList, pi_cast<char *>(MemObj->getZeHandle()) + MapInfo.Offset,
-      MappedPtr, MapInfo.Size, ZeEvent, 0, nullptr));
+  bool deviceIsIntegrated = Queue->Device->ZeDeviceProperties.flags &
```

Comment:
I think it is not valid check to know where memory was allocated. Should either remember that in MemObj, or query with zeMemGetAllocProperties (see example in `piextUSMFree`)

Reply:
The intent is that for integrated devices bufferCreate will allocate host memory and for discrete it will allocate device memory. So this check for whether the device is integrated or discrete, if used consistently in bufferCreate, Map and Umap should work, right?

Reply:
It would work if all context were with single device. Here, we don't know the context, just the device, hence I think we should do what I suggested to know true way how memory was allocated.